[WIP] infra: add k8s-infra-broadcom-admins and terraform for vSphere based CI

chrischdi opened the pull request:
This PR gives a starting point for the provisioning of vSphere based community infrastructure for CI.

upodroid commented:
This is looking good, do you want to apply it?

With regards to the networking, can you create a peering between the vmware network and the default network in k8s-infra-prow-build gcp project? We don't need a separate build cluster if you connect the networks.

```terraform
resource "google_vmwareengine_network_peering" "prow_peering" {
    name = "peer-with-k8s-infra-prow-build"
    description = "Peering with prow build cluster"
    peer_network = "projects/k8s-infra-prow-build/global/networks/default"
    peer_network_type = "STANDARD"
    vmware_engine_network = google_vmwareengine_network.vsphere-network.id
}
```

chrischdi commented:
> @chrischdi Beside the emails, I think we want to drop the `broadcom` term. This is technically owned by CNCF and the K8s project.

Fine for me :-) thought it makes sense to connect the dots.

How should we call it so there's no naming clash? `vsphere` or `gcve`?

ameukam commented:
> > @chrischdi Beside the emails, I think we want to drop the `broadcom` term. This is technically owned by CNCF and the K8s project.
> 
> Fine for me :-) thought it makes sense to connect the dots.
> 
> How should we call it so there's no naming clash? `vsphere` or `gcve`?

my suggestion would be `gcp-gcve` so we connect the dots between platforms and products but not a strong suggestion. :-)

chrischdi commented:
Updated to use gcp-gcve everywhere :-)

Also updated the terraform config how it could work for the gcve cluster, for now as the limited 1 node option.

sbueringer commented on infra/gcp/terraform/k8s-infra-gcp-gcve/README.md:
```diff
@@ -0,0 +1,74 @@
+
+* `k8s-gcp-gcve-ci-username` with value `prow-ci-user@gve.local`
+* `k8s-gcp-gcve-ci-password` with value set above for `GCVE_PROW_CI_PASSWORD`
+* `k8s-gcp-gcve-ci-url` with value set above for `GOVC_SERVER`
```
```suggestion
* `k8s-gcp-gcve-ci-url` with value set above for `GOVC_URL`
```

sbueringer commented on infra/gcp/terraform/k8s-infra-gcp-gcve/main.tf:
```diff
@@ -0,0 +1,50 @@
+}
+
+# TODO(chrischdi): we first need the group
+# // Ensure k8s-infra-vsphere@kubernetes.io has owner access to this project
```
Should this be: "k8s-infra-gcp-gcve-admins@kubernetes.io"?

chrischdi replied:
absolutely :-)

sbueringer commented on infra/gcp/terraform/k8s-infra-gcp-gcve/maintenance-jumphost/README.md:
````diff
@@ -0,0 +1,64 @@
+export CLIENT_PUBLIC_KEY="$(echo $CLIENT_PRIVATE_KEY | wg pubkey)"
+```
+
+Next we have to pick an free IP address from the `192.168.29.0/24` subnet, which is not already in the server config file (see gcloud secret `maintenance-vm-wireguard-config`):
````
```suggestion
Next we have to pick a free IP address from the `192.168.29.0/24` subnet, which is not already in the server config file (see gcloud secret `maintenance-vm-wireguard-config`):
```
nit

sbueringer commented on infra/gcp/terraform/k8s-infra-gcp-gcve/maintenance-jumphost/jumphost.tf:
```diff
@@ -0,0 +1,52 @@
+    }
+  }
+
+  # can_ip_forward = true
```
use/delete? (Fine to keep as is if you want)

sbueringer commented on infra/gcp/terraform/k8s-infra-gcp-gcve/vsphere/modules/gcp-gcve-project/main.tf:
```diff
@@ -0,0 +1,66 @@
+
+# Permissions
+
+# resource "vsphere_entity_permissions" "permission_resource_pool_project" {
```
Should we add a TODO here?

chrischdi replied:
Removing it for now. Have to follow-up on it once we have a job to work with.

TL/DR is: the terraform plugin has a bug which leads to we may not be able to use it at all.

sbueringer commented on infra/gcp/terraform/k8s-infra-gcp-gcve/vsphere/modules/gcp-gcve-project/variables.tf:
```diff
@@ -0,0 +1,50 @@
+*/
+
+variable "project_name" {
+  description = "The name of the project which normally matches the repository name, e.g. cluster-api-provider-vsphere"
```
Is this still correct? (similar below)

sbueringer commented on infra/gcp/terraform/k8s-infra-gcp-gcve/vsphere/projects.tf:
```diff
@@ -0,0 +1,49 @@
+  gcp_gcve_project_name = "k8s-infra-e2e-gcp-gcve-project"
+}
+
+# ## create global permissions to read cluster, datastore, host information etc.
```
TODO?

sbueringer commented on infra/gcp/terraform/k8s-infra-gcp-gcve/vsphere/scripts/boskos-userdata.sh:
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
```
I assume we'll document how this is used later? (If I see correctly it's not documented yet)

chrischdi replied:
Yes. It is not used yet, just helps to get the next step started.

sbueringer reviewed (commented):
Just a few nits

sbueringer commented:
Thx! lgtm +/- CI

sbueringer commented:
/lgtm
